src/widgets/scrollbar.rs

```diff
@@ -446,198 +450,203 @@ impl<'a> Scrollbar<'a> {
-            } else {
-                // For horizontal scrollbar, reduce the width by one
-                Rect::new(area.x, area.y, area.width.saturating_sub(1), area.height)
-            }
-        } else {
-            area
+            area.width
         }
     }
 
-    fn should_not_render(&self, track_start: u16, track_end: u16, content_length: usize) -> bool {
-        if track_end - track_start == 0 || content_length == 0 {
-            return true;
+    fn get_scollbar_area(&self, area: Rect) -> Rect {
+        match self.orientation {
+            ScrollbarOrientation::VerticalLeft => Rect { width: 1, ..area },
+            ScrollbarOrientation::VerticalRight => Rect {
+                x: area.right().saturating_sub(1),
+                width: 1,
+                ..area
+            },
+            ScrollbarOrientation::HorizontalTop => Rect { height: 1, ..area },
+            ScrollbarOrientation::HorizontalBottom => Rect {
+                y: area.bottom().saturating_sub(1),
+                height: 1,
+                ..area
+            },
         }
-        false
     }
 
-    fn get_track_start_end(&self, area: Rect) -> (u16, u16, u16) {
-        match self.orientation {
-            ScrollbarOrientation::VerticalRight => {
-                (area.top(), area.bottom(), area.right().saturating_sub(1))
-            }
-            ScrollbarOrientation::VerticalLeft => (area.top(), area.bottom(), area.left()),
-            ScrollbarOrientation::HorizontalBottom => {
-                (area.left(), area.right(), area.bottom().saturating_sub(1))
-            }
-            ScrollbarOrientation::HorizontalTop => (area.left(), area.right(), area.top()),
-        }
+    /// Calculates length of the track without excluding the arrow heads
+    /// ```plain
+    ///        ┌────────── track_length
+    ///  vvvvvvvvvvvvvvv
+    /// <═══█████═══════>
+    /// ```
+    ///
+    /// # Returns
+    ///
+    /// `(track_length)`
+    fn get_track_length(&self, area: Rect) -> u16 {
+        let (mut track_start, mut track_end) = match self.orientation {
+            ScrollbarOrientation::VerticalRight => (area.y, (area.y + area.height)),
+            ScrollbarOrientation::VerticalLeft => (area.y, (area.y + area.height)),
+            ScrollbarOrientation::HorizontalBottom => (area.x, (area.x + area.width)),
+            ScrollbarOrientation::HorizontalTop => (area.x, (area.x + area.width)),
+        };
+        // if scrollbar has begin and end symbols:
+        //
+        // <═══█████═══════>
+        //
+        // then increment and decrement track_start and track_end respectively
+        if let Some(s) = self.begin_symbol {
+            track_start = track_start.saturating_add(s.width() as u16);
+        };
+        if let Some(s) = self.end_symbol {
+            track_end = track_end.saturating_sub(s.width() as u16);
+        };
+        track_end.saturating_sub(track_start)
     }
```

Comment:
```suggestion
    /// Calculates length of the track excluding the arrow heads
    /// ```plain
    ///        ┌────────── track_length
    ///  vvvvvvvvvvvvvvv
    /// <═══█████═══════>
    /// ```
    ///
    /// # Returns
    ///
    /// `(track_length)`
    fn track_length(&self, area: Rect) -> u16 {
        let start_len = self.begin_symbol.map(|s| s.width() as u16).unwrap_or(0);
        let end_len = self.end_symbol.map(|s| s.width() as u16).unwrap_or(0);
        let arrows_len = start_len + end_len;
        if self.orientation.is_vertical() {
            area.height.saturating_sub(arrows_len)
        } else {
            area.width.saturating_sub(arrows_len)
        }
    }
```